### PR TITLE
[solver,Implicit] Clean up computation of right hand side

### PR DESCRIPTION
**Pull request by florent-lamiraux**
- Methods acting on right hand side of Implicit are now deprecated,
  - Implicit::rightHandSideFromConfig now takes the resulting right hand side as a parameter.
  - Computation of right hand side in solvers have been moved to Implicit class.

Tested on benchmarks 'romeo-placard' and 'ur3-spheres'.

### CODE REVIEW

File: src/solver/hierarchical-iterative.cc [hunk 1]
```diff
@@ -468,24 +449,19 @@ namespace hpp {
           return false;
         }
         size_type iq = iq_ [f];
-        size_type iv = iv_ [f];
         size_type nq = space->nq ();
+#ifndef NDEBUG
```
Comment (jmirabel) [inline src/solver/hierarchical-iterative.cc:453]: I guess this was unintentionally staged.

**Review by jmirabel — COMMENTED** [review]
I still don't understand how your editor handles indentation. In some places, it removed tabs by spaces and in other it used tabs...

File: src/solver/hierarchical-iterative.cc (hunk 1 above)
Comment (florent-lamiraux) [reply src/solver/hierarchical-iterative.cc:453]: No, there is an assert a few lines later.

Comment (jmirabel) [reply src/solver/hierarchical-iterative.cc:453]: Oh, yes. Sorry.

**Review by jmirabel — APPROVED** [review]

**Pull request merged**